lib/std/os/test.zig

```diff
@@ -646,3 +646,34 @@ test "shutdown socket" {
     };
     os.closeSocket(sock);
 }
+
+var signal_test_failed = true;
+
+test "sigaction" {
+    if (builtin.os.tag == .wasi or builtin.os.tag == .windows)
+        return error.SkipZigTest;
+
+    const S = struct {
+        fn handler(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const c_void) callconv(.C) void {
+            // Check that we received the correct signal.
+            if (sig == os.SIGUSR1 and sig == info.signo)
```

Comment:
after testing on OpenBSD, `info.signo` doesn't work here: `error: no member named 'signo' in union 'std.os.bits.openbsd.siginfo_t'`

but it seems the `siginfo_t` definition for openbsd could be improved to match it. it could be addressed separately